osu.Game/Screens/SelectV2/BeatmapCarouselFilterSorting.cs

```diff
@@ -36,7 +36,7 @@ public async Task<IEnumerable<CarouselItem>> Run(IEnumerable<CarouselItem> items
                 switch (criteria.Sort)
                 {
                     case SortMode.Artist:
-                        comparison = OrdinalSortByCaseStringComparer.DEFAULT.Compare(ab.BeatmapSet!.Metadata.Artist, bb.BeatmapSet!.Metadata.Artist);
+                        comparison = OrdinalSortByCaseStringComparer.DEFAULT.Compare(ab.Metadata.Artist, bb.Metadata.Artist);
```

Comment:
I dunno about these changes... Metadata is technically not supposed to vary within a single set, but it could, and I don't think we check that in any way. Could lead to some dumb shenanigans happening.

At least previously, because of going through the set, it was assured that every beatmap of a set had the same metadata (namely the metadata of the first beatmap in the set).

Reply:
The overhead of the set traversal was quite large (in the seconds for the performance test).

I don't see this change as an issue because if/when we change the correlation of metadata to sets/beatmaps, it's a given that we'd be examining every usage with a magnifying glass.